diff --git a/frameworks/core/components_ng/pattern/text_field/text_field_pattern.cpp b/frameworks/core/components_ng/pattern/text_field/text_field_pattern.cpp
index 68bb65778eb0cb4319ec286fe2a4f7123912ebfb..124e90c71a6d2e80b93adde303b467ea868dbe9f 100644
--- a/frameworks/core/components_ng/pattern/text_field/text_field_pattern.cpp
+++ b/frameworks/core/components_ng/pattern/text_field/text_field_pattern.cpp
@@ -821,13 +821,8 @@ void TextFieldPattern::UpdateCaretInfoToController(bool forceUpdate)
 {
     CHECK_NULL_VOID(HasFocus());
 #if defined(CROSS_PLATFORM)
-#if defined(IOS_PLATFORM)
     if (editingValue_ && editingValue_->selection.IsValid() &&
         editingValue_->selection.GetEnd() < selectController_->GetCaretIndex()) {
-#else
-    if (editingValue_ && editingValue_->selection.IsValid() &&
-        editingValue_->selection.GetEnd() < selectController_->GetCaretIndex() && !editingValue_->appendText.empty()) {
-#endif
         SetCaretPosition(editingValue_->selection.GetEnd());
     }
     if (editingValue_ && editingValue_->selection.IsValid()) {
@@ -5214,7 +5209,13 @@ void TextFieldPattern::ExecuteInsertValueCommand(const InsertCommandInfo& info)
         CalcCounterAfterFilterInsertValue(originLength, insertValue,
             static_cast<int32_t>(layoutProperty->GetMaxLengthValue(Infinity<uint32_t>())));
     }
+#ifdef ANDROID_PLATFORM
+    if (info.textSelection.baseOffset == info.textSelection.extentOffset) {
+        selectController_->UpdateCaretIndex(info.textSelection.baseOffset);
+    }
+#else
     selectController_->UpdateCaretIndex(caretStart + caretMoveLength);
+#endif
     UpdateObscure(insertValue, hasInsertValue);
     UpdateEditingValueToRecord();
     if (isIME) {
@@ -6030,11 +6031,6 @@ bool TextFieldPattern::HandleEditingEventCrossPlatform(const std::shared_ptr<Tex
     }
 #endif
     editingValue_ = value;
-#ifdef ANDROID_PLATFORM
-    if (value->appendText.empty()) {
-        return true;
-    }
-#endif
     InsertValue(UtfUtils::Str8DebugToStr16(value->appendText), true);
     return true;
 #endif // CROSS_PLATFORM
@@ -10953,6 +10949,7 @@ void TextFieldPattern::AddInsertCommand(const std::u16string& insertValue, Input
         info.compose.end = editingValue_->compose.GetEnd();
         info.compose.isActive = editingValue_->compose.IsValid();
         info.unmarkText = editingValue_->unmarkText;
+        info.textSelection = editingValue_->selection;
     }
 #endif
     insertCommands_.emplace(info);
diff --git a/frameworks/core/components_ng/pattern/text_field/text_field_pattern.h b/frameworks/core/components_ng/pattern/text_field/text_field_pattern.h
index 87162dfdd8fd2dbbf56828a474f2ffeccfbfd82f..c244110d3933d4b2e56154252db970096f71b1af 100644
--- a/frameworks/core/components_ng/pattern/text_field/text_field_pattern.h
+++ b/frameworks/core/components_ng/pattern/text_field/text_field_pattern.h
@@ -206,6 +206,7 @@ struct InsertCommandInfo {
 #if defined(CROSS_PLATFORM)
     InsertCommandComposeInfo compose;
     bool unmarkText;
+    TextSelection textSelection;
 #endif
 };
 
diff --git a/test/unittest/core/pattern/text_input/text_field_pattern_test.cpp b/test/unittest/core/pattern/text_input/text_field_pattern_test.cpp
index 08c57e3d3be7a9ea40a2c693e01d3003d817d9cb..d8d3a0eee82cbc335453f1fa802565bf96d3db22 100644
--- a/test/unittest/core/pattern/text_input/text_field_pattern_test.cpp
+++ b/test/unittest/core/pattern/text_input/text_field_pattern_test.cpp
@@ -2942,6 +2942,12 @@ HWTEST_F(TextFieldPatternTest, TextFieldShiftMultipleSelection001, TestSize.Leve
     pattern->HandleMouseEvent(info);
     pattern->UpdateCaretByClick(offset);
 
+    /**
+     * 
+     * @tc.step:
+     * check if pattern->IsSelected is false
+     *
+     */
     EXPECT_EQ(pattern->IsSelected(), false);
 }
 
