diff --git a/frameworks/core/components_ng/pattern/overlay/overlay_manager.cpp b/frameworks/core/components_ng/pattern/overlay/overlay_manager.cpp
index 98922c4fad1c4f8c708c95f562932cca2fac2b95..0df0f6d13a6a00f7370c4de5466b2e21279d5a2c 100644
--- a/frameworks/core/components_ng/pattern/overlay/overlay_manager.cpp
+++ b/frameworks/core/components_ng/pattern/overlay/overlay_manager.cpp
@@ -2914,7 +2914,9 @@ void OverlayManager::HideMenuInSubWindow(bool showPreviewAnimation, bool startDr
         return;
     }
     auto rootNode = rootNodeWeak_.Upgrade();
-    for (const auto& child : rootNode->GetChildren()) {
+    CHECK_NULL_VOID(rootNode);
+    auto children = rootNode->GetChildren();
+    for (const auto& child : children) {
         auto node = DynamicCast<FrameNode>(child);
         if (node->GetTag() == V2::MENU_WRAPPER_ETS_TAG) {
             PopMenuAnimation(node, showPreviewAnimation, startDrag);
diff --git a/test/unittest/core/pattern/overlay/overlay_new_test_ng.cpp b/test/unittest/core/pattern/overlay/overlay_new_test_ng.cpp
index 34a1c363e4af57ac31c04e077d17ba3569c7685d..b3c4b013d12759db584bc56b77052fadc499d275 100644
--- a/test/unittest/core/pattern/overlay/overlay_new_test_ng.cpp
+++ b/test/unittest/core/pattern/overlay/overlay_new_test_ng.cpp
@@ -1522,6 +1522,10 @@ HWTEST_F(OverlayNewTestNg, MenuNewTest018, TestSize.Level1)
     EXPECT_TRUE(overlayManager->menuMap_.empty());
     auto layoutProperty = menuNode->GetLayoutProperty();
     CHECK_NULL_VOID(layoutProperty);
+    /**
+     * @tc.steps: step3. set visibility gone and call ShowMenuAnimation.
+     * @tc.expected: visibility will change to VISIBLE.
+     */
     layoutProperty->UpdateVisibility(VisibleType::GONE, true);
     overlayManager->ShowMenuAnimation(menuNode);
     overlayManager->UpdateMenuVisibility(menuNode);
@@ -1534,7 +1538,7 @@ HWTEST_F(OverlayNewTestNg, MenuNewTest018, TestSize.Level1)
     EXPECT_TRUE(menuHelper);
 
     /**
-     * @tc.steps: step3. call HideMenu related functions after menuNode already erased.
+     * @tc.steps: step4. call HideMenu related functions after menuNode already erased.
      * @tc.expected: return normally
      */
     overlayManager->HideAllMenus();
@@ -1545,7 +1549,7 @@ HWTEST_F(OverlayNewTestNg, MenuNewTest018, TestSize.Level1)
     EXPECT_TRUE(overlayManager->menuMap_.empty());
 
     /**
-     * @tc.steps: step4. call DeleteMenu again after menuNode already erased.
+     * @tc.steps: step5. call DeleteMenu again after menuNode already erased.
      * @tc.expected: return normally
      */
     overlayManager->RemoveMenu(menuNode);
