diff --git a/icu4c/source/tools/genrb/parse.cpp b/icu4c/source/tools/genrb/parse.cpp
index 2c5d49524369b225be756802a34fbcceb6b9c7c0..9f5a184be4e7de1b2ce2dae8d601fea3dd319a8b 100644
--- a/icu4c/source/tools/genrb/parse.cpp
+++ b/icu4c/source/tools/genrb/parse.cpp
@@ -1153,7 +1153,9 @@ addCollation(ParseState* state, TableResource  *result, const char *collationTyp
     struct UString    *tokenValue;
     struct UString     comment;
     enum   ETokenType  token;
-    char               subtag[1024];
+    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+    CharString         subtag;
+    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
     UnicodeString      rules;
     UBool              haveRules = false;
     UVersionInfo       version;
@@ -1188,17 +1190,18 @@ addCollation(ParseState* state, TableResource  *result, const char *collationTyp
 
             return NULL;
         }
-
-        u_UCharsToChars(tokenValue->fChars, subtag, u_strlen(tokenValue->fChars) + 1);
-
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+        subtag.clear();
+        subtag.appendInvariantChars(tokenValue->fChars, u_strlen(tokenValue->fChars), *status);
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
         if (U_FAILURE(*status))
         {
             res_close(result);
             return NULL;
         }
-
-        member = parseResource(state, subtag, NULL, status);
-
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+        member = parseResource(state, subtag.data(), nullptr, status);
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
         if (U_FAILURE(*status))
         {
             res_close(result);
@@ -1208,7 +1211,9 @@ addCollation(ParseState* state, TableResource  *result, const char *collationTyp
         {
             // Ignore the parsed resources, continue parsing.
         }
-        else if (uprv_strcmp(subtag, "Version") == 0 && member->isString())
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+        else if (uprv_strcmp(subtag.data(), "Version") == 0 && member->isString())
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
         {
             StringResource *sr = static_cast<StringResource *>(member);
             char     ver[40];
@@ -1225,11 +1230,15 @@ addCollation(ParseState* state, TableResource  *result, const char *collationTyp
             result->add(member, line, *status);
             member = NULL;
         }
-        else if(uprv_strcmp(subtag, "%%CollationBin")==0)
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+        else if(uprv_strcmp(subtag.data(), "%%CollationBin")==0)
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
         {
             /* discard duplicate %%CollationBin if any*/
         }
-        else if (uprv_strcmp(subtag, "Sequence") == 0 && member->isString())
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+        else if (uprv_strcmp(subtag.data(), "Sequence") == 0 && member->isString())
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
         {
             StringResource *sr = static_cast<StringResource *>(member);
             rules = sr->fString;
@@ -1395,7 +1404,9 @@ parseCollationElements(ParseState* state, char *tag, uint32_t startline, UBool n
     struct UString    *tokenValue;
     struct UString     comment;
     enum   ETokenType  token;
-    char               subtag[1024], typeKeyword[1024];
+    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+    CharString         subtag, typeKeyword;
+    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
     uint32_t           line;
 
     result = table_open(state->bundle, tag, NULL, status);
@@ -1436,18 +1447,20 @@ parseCollationElements(ParseState* state, char *tag, uint32_t startline, UBool n
 
                 return NULL;
             }
-
-            u_UCharsToChars(tokenValue->fChars, subtag, u_strlen(tokenValue->fChars) + 1);
-
+            /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+            subtag.clear();
+            subtag.appendInvariantChars(tokenValue->fChars, u_strlen(tokenValue->fChars), *status);
+            /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
             if (U_FAILURE(*status))
             {
                 res_close(result);
                 return NULL;
             }
-
-            if (uprv_strcmp(subtag, "default") == 0)
+            /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+            if (uprv_strcmp(subtag.data(), "default") == 0)
             {
-                member = parseResource(state, subtag, NULL, status);
+                member = parseResource(state, subtag.data(), nullptr, status);
+            /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
 
                 if (U_FAILURE(*status))
                 {
@@ -1466,22 +1479,35 @@ parseCollationElements(ParseState* state, char *tag, uint32_t startline, UBool n
                 if(token == TOK_OPEN_BRACE) {
                     token = getToken(state, &tokenValue, &comment, &line, status);
                     TableResource *collationRes;
-                    if (keepCollationType(subtag)) {
-                        collationRes = table_open(state->bundle, subtag, NULL, status);
+                    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+                    if (keepCollationType(subtag.data())) {
+                        collationRes = table_open(state->bundle, subtag.data(), nullptr, status);
+                    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
                     } else {
                         collationRes = NULL;
                     }
                     // need to parse the collation data regardless
-                    collationRes = addCollation(state, collationRes, subtag, startline, status);
+                    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+                    collationRes = addCollation(state, collationRes, subtag.data(), startline, status);
+                    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
                     if (collationRes != NULL) {
                         result->add(collationRes, startline, *status);
                     }
                 } else if(token == TOK_COLON) { /* right now, we'll just try to see if we have aliases */
                     /* we could have a table too */
                     token = peekToken(state, 1, &tokenValue, &line, &comment, status);
-                    u_UCharsToChars(tokenValue->fChars, typeKeyword, u_strlen(tokenValue->fChars) + 1);
-                    if(uprv_strcmp(typeKeyword, "alias") == 0) {
-                        member = parseResource(state, subtag, NULL, status);
+                    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+                    typeKeyword.clear();
+                    typeKeyword.appendInvariantChars(tokenValue->fChars, u_strlen(tokenValue->fChars), *status);
+                    if (U_FAILURE(*status))
+                    {
+                        res_close(result);
+                        return nullptr;
+                    }
+
+                    if(uprv_strcmp(typeKeyword.data(), "alias") == 0) {
+                        member = parseResource(state, subtag.data(), nullptr, status);
+                        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
                         if (U_FAILURE(*status))
                         {
                             res_close(result);
@@ -1523,7 +1549,9 @@ realParseTable(ParseState* state, TableResource *table, char *tag, uint32_t star
     struct UString    *tokenValue=NULL;
     struct UString    comment;
     enum   ETokenType token;
-    char              subtag[1024];
+    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+    CharString        subtag;
+    /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
     uint32_t          line;
     UBool             readToken = false;
 
@@ -1562,7 +1590,10 @@ realParseTable(ParseState* state, TableResource *table, char *tag, uint32_t star
         }
 
         if(uprv_isInvariantUString(tokenValue->fChars, -1)) {
-            u_UCharsToChars(tokenValue->fChars, subtag, u_strlen(tokenValue->fChars) + 1);
+            /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+            subtag.clear();
+            subtag.appendInvariantChars(tokenValue->fChars, u_strlen(tokenValue->fChars), *status);
+            /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
         } else {
             *status = U_INVALID_FORMAT_ERROR;
             error(line, "invariant characters required for table keys");
@@ -1574,9 +1605,9 @@ realParseTable(ParseState* state, TableResource *table, char *tag, uint32_t star
             error(line, "parse error. Stopped parsing tokens with %s", u_errorName(*status));
             return NULL;
         }
-
-        member = parseResource(state, subtag, &comment, status);
-
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 begin */
+        member = parseResource(state, subtag.data(), &comment, status);
+        /* <issue: https://github.com/unicode-org/icu/pull/3340> 20250912 end */
         if (member == NULL || U_FAILURE(*status))
         {
             error(line, "parse error. Stopped parsing resource with %s", u_errorName(*status));
